ERROR: conversion between UTF8 and SQL_ASCII is not supported

I'm splitting this off from #8272 because @landreev and I have been discussing the possibility of wrapping src/main/resources/db/migration/V5.8.0.2__8018-invalid-characters.sql from #8242 into some logic to only run on databases that have encoding of `UTF8`.

That is, as explained in the other issue, if databases have an encoding of `SQL_ASCII` like this...

```
postgres=# \l
                             List of databases
   Name    |  Owner   | Encoding  | Collate | Ctype |   Access privileges   
-----------+----------+-----------+---------+-------+-----------------------
 dvndb     | dvnapp   | SQL_ASCII | C       | C     | 
 postgres  | postgres | SQL_ASCII | C       | C     | 
 template0 | postgres | SQL_ASCII | C       | C     | =c/postgres          +
           |          |           |         |       | postgres=CTc/postgres
 template1 | postgres | SQL_ASCII | C       | C     | =c/postgres          +
           |          |           |         |       | postgres=CTc/postgres
(4 rows)
```

... we expect deployment to fail with this in server.log:

```
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V5.8.0.2__8018-invalid-characters.sql failed
------------------------------------------------------
SQL State  : 0A000
Error Code : 0
Message    : ERROR: conversion between UTF8 and SQL_ASCII is not supported
  Position: 110
Location   : db/migration/V5.8.0.2__8018-invalid-characters.sql (/opt/payara5/glassfish/domains/domain1/applications/dataverse/WEB-INF/classes/db/migration/V5.8.0.2__8018-invalid-characters.sql)
Line       : 4
Statement  : -- Remove character: start of text (\u0002)
UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\0002', '', 'g');
-- Remove character: not a character (\ufffe)
UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\FFFE', '', 'g')
```

--- a/src/main/resources/db/migration/V5.8.0.2__8018-invalid-characters.sql
+++ b/src/main/resources/db/migration/V5.8.0.2__8018-invalid-characters.sql
@@ -1,7 +1,14 @@
--- This list of characters also appears in DatasetFieldValidator.java
--- Remove character: form feed (\f)
-UPDATE datasetfieldvalue SET value = regexp_replace(value, E'\f', '', 'g');
--- Remove character: start of text (\u0002)
-UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\0002', '', 'g');
--- Remove character: not a character (\ufffe)
-UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\FFFE', '', 'g');
+DO $$
+BEGIN
+-- Run on UTF8 only. It's known to fail on SQL_ASCII encoding.
+IF EXISTS (SELECT 1 FROM information_schema.character_sets WHERE character_set_name='UTF8') THEN
+  -- This list of characters also appears in DatasetFieldValidator.java
+  -- Remove character: form feed (\f)
+  UPDATE datasetfieldvalue SET value = regexp_replace(value, E'\f', '', 'g');
+  -- Remove character: start of text (\u0002)
+  UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\0002', '', 'g');
+  -- Remove character: not a character (\ufffe)
+  UPDATE datasetfieldvalue SET value = regexp_replace(value, U&'\FFFE', '', 'g');
+END IF;
+END
+$$

